libpod/image/image.go

```diff
@@ -149,43 +210,344 @@ func (i *Image) ID() string {
 	return i.image.ID
 }
 
-// createNamesToPull looks at a decomposed image and determines the possible
-// images names to try pulling in combination with the registries.conf file as well
-func (i *Image) createNamesToPull() ([]string, error) {
-	var pullNames []string
-	decomposedImage, err := decompose(i.InputName)
+// Digest returns the image's Manfiest
+func (i *Image) Digest() digest.Digest {
+	return i.image.Digest
+}
+
+// Names returns a string array of names associated with the image
+func (i *Image) Names() []string {
+	return i.image.Names
+}
+
+// Created returns the time the image was created
+func (i *Image) Created() time.Time {
+	return i.image.Created
+}
+
+// TopLayer returns the top layer id as a string
+func (i *Image) TopLayer() string {
+	return i.image.TopLayer
+}
+
+// Remove an image; container removal for the image must be done
+// outside the context of images
+func (i *Image) Remove(force bool) error {
+	_, err := i.imageruntime.store.DeleteImage(i.ID(), true)
+	return err
+}
+
+func (i *Image) GetImageInspectInfo() (*inspect.ImageData, error) {
+
+	layer, err := i.imageruntime.store.Layer(i.TopLayer())
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading information about layer %q", i.TopLayer())
+	}
+	// TODO: I think this might the wrong way to get this? double check it
+	size, err := i.imageruntime.store.DiffSize(layer.Parent, layer.ID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error determining size of layer %q", layer.ID)
+	}
+	driverData, err := driver.GetDriverData(i.imageruntime.store, layer.ID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting graph driver info %q", i.ID())
+	}
+	return i.getImageData(size, driverData)
+}
+
+func (i *Image) getImageData(size int64, driver *inspect.Data) (*inspect.ImageData, error) {
+	imgRef, err := i.toImageRef()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading image %q", i.ID())
+	}
+
+	imgSize, err := imgRef.Size()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading size of image %q", i.ID())
+	}
+	manifest, manifestType, err := imgRef.Manifest()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading manifest for image %q", i.ID())
+	}
+	imgDigest := digest.Digest("")
+	if len(manifest) > 0 {
+		imgDigest = digest.Canonical.FromBytes(manifest)
+	}
+	annotations := annotations(manifest, manifestType)
+
+	ociv1Img, err := imgRef.OCIConfig()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting oci image info %q", i.ID())
+	}
+	info, err := imgRef.Inspect()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting image info %q", i.ID())
+	}
+
+	var repoDigests []string
+	for _, name := range i.Names() {
+		repoDigests = append(repoDigests, strings.SplitN(name, ":", 2)[0]+"@"+imgDigest.String())
+	}
+
+	data := &inspect.ImageData{
```

Comment:
Don't we have this struct in our new Image struct? Can we use that to cache this?

Reply:
removed, should be in runtime_img.go